### PR TITLE
refactor(lane_change): refactor code using transient data

### PR DESCRIPTION
**Pull request by mkquda**
## Description

Recently, and new data struct `TransientData` was added to standardize commonly used values in the lane change module.

This PR refactors lane change code using `TransientData` to improve code quality and readability, and reduce complexity.

## Changes

- Add new member `target_lane_length` to `TransientData` struct
- Add new member `target_lanes_ego_arc` to `TransientData` struct
- Add new member `current_lanes_ego_arc` to `TransientData` struct
- Add new function `get_prepare_metrics()` to reduce size of `get_lane_change_paths()` function and improve granularity
- Add new function `get_lane_changing_metrics()` to reduce size of `get_lane_change_paths()` function and improve granularity
- Add new function `is_valid_prepare_segment()` to reduce size of `get_lane_change_paths()` function and improve granularity

## Related links

None.

## How was this PR tested?
- PSIM
- [TIER IV Internal Evaluator](https://evaluation.tier4.jp/evaluation/reports/ffd1e9d0-69bb-5b07-a8e7-1af75b63d433?page=1&project_id=prd_jt)

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by maxime-clem — APPROVED** [review]

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp [hunk 1]
```diff
@@ -1565,16 +1554,44 @@ bool NormalLaneChange::get_lane_change_paths(LaneChangePaths & candidate_paths)
...
   const LaneChangePhaseMetrics & prep_metrics, const LaneChangePhaseMetrics & lc_metrics,
   const PathWithLaneId & prep_segment, const std::vector<std::vector<int64_t>> & sorted_lane_ids,
-  const Pose & lc_start_pose, const double target_lane_length, const double shift_length,
-  const double next_lc_buffer, const bool is_goal_in_route) const
+  const Pose & lc_start_pose, const double shift_length) const
 {
   const auto & route_handler = *getRouteHandler();
   const auto & current_lanes = common_data_ptr_->lanes_ptr->current;
   const auto & target_lanes = common_data_ptr_->lanes_ptr->target;
+  const auto & target_lane_length = common_data_ptr_->transient_data.target_lane_length;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp:1591]:
```suggestion
  const auto target_lane_length = common_data_ptr_->transient_data.target_lane_length;
```
trivial type, so reference is not needed.

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp [hunk 2]
```diff
@@ -1565,16 +1554,44 @@ bool NormalLaneChange::get_lane_change_paths(LaneChangePaths & candidate_paths)
...
   const auto & forward_path_length = planner_data_->parameters.forward_path_length;
+  const auto & is_goal_in_route = common_data_ptr_->lanes_ptr->target_lane_in_goal_section;
+  const auto & next_lc_buffer = common_data_ptr_->transient_data.next_dist_buffer.min;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp:1594]:
```suggestion
  const auto is_goal_in_route = common_data_ptr_->lanes_ptr->target_lane_in_goal_section;
  const auto next_lc_buffer = common_data_ptr_->transient_data.next_dist_buffer.min;
```
trivial type, so reference is not needed.

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp [hunk 3]
```diff
@@ -1614,8 +1631,9 @@ LaneChangePath NormalLaneChange::get_candidate_path(
 
 bool NormalLaneChange::check_candidate_path_safety(
   const LaneChangePath & candidate_path, const lane_change::TargetObjects & target_objects,
-  const double lane_change_buffer, const bool is_stuck) const
+  const bool is_stuck) const
 {
+  const auto & lane_change_buffer = common_data_ptr_->transient_data.current_dist_buffer.min;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp:1636]:
```suggestion
  const auto lane_change_buffer = common_data_ptr_->transient_data.current_dist_buffer.min;
```
trivial type, so reference is not needed.

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp [hunk 4]
```diff
@@ -1722,6 +1737,7 @@ std::optional<LaneChangePath> NormalLaneChange::calcTerminalLaneChangePath(
   const auto lane_changing_time = PathShifter::calcShiftTimeFromJerk(
     shift_length, lane_change_parameters_->lane_changing_lateral_jerk, max_lateral_acc);
 
+  const auto & target_lane_length = common_data_ptr_->transient_data.target_lane_length;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp:1740]:
```suggestion
  const auto target_lane_length = common_data_ptr_->transient_data.target_lane_length;
```
trivial type, so reference is not needed.

**Review by zulfaqar-azmi-t4 — APPROVED** [review]

**Pull request merged**